src/picomsg/src/oper_db_writer.cpp

```diff
@@ -114,3 +190,282 @@ bool oper_db_writer_set_terminator_vrf_tunnel_counter(const uint32_t terminator_
 
 	return true;
 }
+
+bool oper_db_writer_del_terminator_vrf_tunnel_counter(const uint32_t terminator_id, const char *tenant_id,
+                                                const char *account_id, const char *vrf_name) {
+	auto oper_db_client = picomsg_get_oper_db_client();
+
+	std::string terminator_id_str = std::to_string(terminator_id);
+
+	const char* keys[] = { terminator_id_str.c_str(), tenant_id, account_id, vrf_name };
+
+	auto db_obj = oper_db_client_create_db_object(*oper_db_client, LIBOPER_DB_CLIENT_CLASS_ID_VRF_TUNNELS_NUMBER_IN_TERMINATOR, keys);
+
+	if (! (*oper_db_client)->del((*oper_db_client).get(), db_obj.release(), 0)) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to delete LIBOPER_DB_CLIENT_CLASS_ID_VRF_TUNNELS_NUMBER_IN_TERMINATOR keys {%u, %s, %s, %s} from oper DB",
+		terminator_id, tenant_id, account_id, vrf_name);
+		return false;
+	}
+
+	return true;
+}
+
+bool oper_db_writer_set_terminator_tunnel_oper_data(const struct terminator_tunnel_oper_data *oper_data, bool write_uptime) {
+	auto oper_db_client = picomsg_get_oper_db_client();
+	const char* keys[] = {oper_data->vrf_name, oper_data->device_id};
+	enum LIBOPER_DB_CLIENT_FIELD_ENUM_VALUES oper_db_status_value, oper_db_ike_version;
+
+	if(! tunnel_status_to_oper_db(oper_data->status, oper_db_status_value)) {
+		return false;
+	}
+
+	auto db_obj = oper_db_client_create_db_object(*oper_db_client, LIBOPER_DB_CLIENT_CLASS_ID_TUNNEL_ITEM, keys);
+	if (! db_obj) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to create oper DB object for class ID LIBOPER_DB_CLIENT_CLASS_ID_TUNNEL_OPER_DATA");
+		return false;
+	}
+
+	if (! db_obj->string_set(db_obj.get(), LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_VRF_NAME, oper_data->vrf_name)) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to set oper DB object LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_VRF_NAME to %s",
+					oper_data->vrf_name);
+		return false;
+	}
+
+	if (! db_obj->string_set(db_obj.get(), LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_DEVICE_ID, oper_data->device_id)) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to set oper DB object LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_DEVICE_ID to %s",
+					oper_data->device_id);
+		return false;
+	}
+
+	if (! (*oper_db_client)->set((*oper_db_client).get(), db_obj.release(), 0)) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to write vrf[%s] device-id[%s] tunnel oper data to oper DB", oper_data->vrf_name, oper_data->device_id);
+		return false;
+	}
+
+	db_obj = oper_db_client_create_db_object(*oper_db_client, LIBOPER_DB_CLIENT_CLASS_ID_TUNNEL_OPER_DATA, keys);
```

Comment:
Split this function into 2 - One for `LIBOPER_DB_CLIENT_CLASS_ID_TUNNEL_ITEM` and another for `LIBOPER_DB_CLIENT_CLASS_ID_TUNNEL_OPER_DATA` and have this function call both these helper functions. It's getting hard to keep track of all the fields being set when the `db_obj` changes mid-function